### PR TITLE
docs: fix internal cross-page references by injecting .md

### PR DESCRIPTION
**Pull request by hirenp**
Fixing external references by injecting .md so that links works correctly while reading the documentation.

### CODE REVIEW

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Thanks for this PR! could you update the PR description with a small line detailing the change, so we keep a nice git history.

/ok-to-test

**Comment by hirenp** [comment]
> Something like this should give a good starting point, doing most of the work but possibly doing some incorrect edits and missing some, too:
Thanks! That seemed to have done the trick. I tried spot checking a couple places and they look correct to me.

File: docs/get-started/introduction.md [hunk 1]
```diff
@@ -43,14 +43,14 @@ NFD-Topology-Updater is a daemon responsible for examining allocated
 resources on a worker node to account for resources available to be allocated
 to new pod on a per-zone basis (where a zone can be a NUMA node). It then
 creates or updates a
-[NodeResourceTopology](../usage/custom-resources#noderesourcetopology) custom
+[NodeResourceTopology](../usage/custom-resourcesnoderesourcetopology) custom
```
Comment (marquiz) [inline docs/get-started/introduction.md:46]:
```suggestion
[NodeResourceTopology](../usage/custom-resources.md#noderesourcetopology) custom
```

File: docs/get-started/introduction.md [hunk 2]
```diff
@@ -43,14 +43,14 @@ NFD-Topology-Updater is a daemon responsible for examining allocated
...
 resource object specific to this node. One instance of nfd-topology-updater is
 supposed to be running on each node of the cluster.
 
 ## NFD-Topology-Garbage-Collector
 
 NFD-Topology-Garbage-Collector is a daemon responsible for cleaning obsolete
-[NodeResourceTopology](../usage/custom-resources#noderesourcetopology) objects,
+[NodeResourceTopology](../usage/custom-resourcesnoderesourcetopology) objects,
```
Comment (marquiz) [inline docs/get-started/introduction.md:53]:
```suggestion
[NodeResourceTopology](../usage/custom-resources.md#noderesourcetopology) objects,
```

File: docs/get-started/introduction.md [hunk 3]
```diff
@@ -114,13 +114,13 @@ only created on nodes running nfd-master.
...
 can be used for representing node features and requesting node labels to be
 generated.
 
-NFD-Master uses [NodeFeatureRule](../usage/custom-resources#nodefeaturerule)s
+NFD-Master uses [NodeFeatureRule](../usage/custom-resources.md#nodefeaturerule)s
 for custom labeling of nodes.
 
 NFD-Topology-Updater creates
-[NodeResourceTopology](../usage/custom-resources#noderesourcetopology) objects
+[NodeResourceTopology](../usage/custom-resourcesnoderesourcetopology) objects
```
Comment (marquiz) [inline docs/get-started/introduction.md:125]:
```suggestion
[NodeResourceTopology](../usage/custom-resources.md#noderesourcetopology) objects
```

File: docs/usage/nfd-topology-gc.md [hunk 4]
```diff
@@ -24,6 +24,6 @@ when no -gc-interval is specified.
 ## Topology-Garbage-Collector Configuration
 
 In Helm deployments,
-(see [Topology Garbage Collector](../deployment/helm.md#topology-garbage-collector-parameters)
+(see [Topology Garbage Collector](../deployment/helmtopology-garbage-collector-parameters)
```
Comment (marquiz) [inline docs/usage/nfd-topology-gc.md:27]:
```suggestion
(see [Topology Garbage Collector](../deployment/helm.md#topology-garbage-collector-parameters)
```

File: docs/usage/nfd-topology-updater.md [hunk 5]
```diff
@@ -12,18 +12,18 @@ sort: 5
...
 When run as a daemonset, nodes are re-examined for the allocated resources
 (to determine the information of the allocatable resources on a per zone basis
 where a zone can be a NUMA node) at an interval specified using the
-[`-sleep-interval`](../reference/topology-updater-commandline-reference.html#-sleep-interval)
+[`-sleep-interval`](../reference/topology-updater-commandline-reference.html.md#-sleep-interval)
 option. The default sleep interval is set to 60s which is the value when no
 -sleep-interval is specified.
 In addition, it can avoid examining specific allocated resources
-given a configuration of resources to exclude via [`-excludeList`](../reference/topology-updater-configuration-reference.md#excludelist)
+given a configuration of resources to exclude via [`-excludeList`](../reference/topology-updater-configuration-referenceexcludelist)
```
Comment (marquiz) [inline docs/usage/nfd-topology-updater.md:26]:
```suggestion
given a configuration of resources to exclude via [`-excludeList`](../reference/topology-updater-configuration-reference.md#excludelist)
```

File: docs/usage/nfd-topology-updater.md [hunk 6]
```diff
@@ -60,7 +60,7 @@ kubectl -n ${NFD_NS} edit configmap nfd-topology-updater-conf
 ```
 
 In Helm deployments,
-[Topology Updater parameters](../deployment/helm.md#topology-updater-parameters)
+[Topology Updater parameters](../deployment/helmtopology-updater-parameters)
```
Comment (marquiz) [inline docs/usage/nfd-topology-updater.md:63]:
```suggestion
[Topology Updater parameters](../deployment/helm.md#topology-updater-parameters)
```

File: docs/usage/nfd-worker.md [hunk 7]
```diff
@@ -22,7 +22,7 @@ Worker connects to the nfd-master service to advertise hardware features.
 
 When run as a daemonset, nodes are re-labeled at an default interval of 60s.
 This can be changed by using the
-[`core.sleepInterval`](../reference/worker-configuration-reference.html#coresleepinterval)
+[`core.sleepInterval`](../reference/worker-configuration-reference.html.md#coresleepinterval)
```
Comment (marquiz) [inline docs/usage/nfd-worker.md:25]:
```suggestion
[`core.sleepInterval`](../reference/worker-configuration-reference.md#coresleepinterval)
```

**Review by marquiz — COMMENTED** [review]
Thanks for the updater @hirenp. I found some mistakes, possibly because of erroneous sed script that I provided if you used them.

Could you also squash the PR into one commit and adjust the commit message accordingly, e.g. `"docs: fix internal cross-page references by injecting .md"` or smth

**Comment by hirenp** [comment]
> Thanks for the updater @hirenp. I found some mistakes, possibly because of erroneous sed script that I provided if you used them.
> 
> Could you also squash the PR into one commit and adjust the commit message accordingly, e.g. `"docs: fix internal cross-page references by injecting .md"` or smth

Apologies for missing those. I fixed as per your suggestions.

**Comment by marquiz** [comment]
> Apologies for missing those. I fixed as per your suggestions.

Np, thanks for the update @hirenp. Looks good, now. Just one last nit: could you remove the inadvertent quotation mark (`"`, a copy-paste mistake(?))  from the end of the commit message

/retitle docs: fix internal cross-page references by injecting .md

**Comment by hirenp** [comment]
> > Apologies for missing those. I fixed as per your suggestions.
> 
> Np, thanks for the update @hirenp. Looks good, now. Just one last nit: could you remove the inadvertent quotation mark (`"`, a copy-paste mistake(?)) from the end of the commit message

:man_facepalming: 

Fixed. Thanks.

**Review by marquiz — APPROVED** [review]
👍

**Review by fmuyassarov — COMMENTED** [review]
/lgtm

**Pull request merged**